trestle/core/resolver/_import.py

```diff
@@ -50,6 +52,20 @@ def __init__(
         self._change_prose = change_prose
         self._params_format = params_format
         self._param_rep = param_rep
+        self._resources = resources
+
+        if self._import.href[0] == '#':
+            # Specification section on internal reference resolution:
+            # https://pages.nist.gov/OSCAL/concepts/processing/profile-resolution/#d2e300-head
+            try:
+                resource = [r for r in self._resources if r.uuid == self._import.href[1:]][0]
+                self._import.href = [rlink.href for rlink in resource.rlinks if rlink.href.endswith('.json')][0]
+
```

Comment:
We are trying to support both json and yaml so please include a fallback to yaml if json is not available.  I don't know if any relevant resources are yaml currently but we want to support it if possible.  If you just find the list of json matches and check if it is empty before indexing i9t you can keep it all in one try block.

Reply:
Is there a way to know a given profile's `FileContentType` by introspecting it, @fsuits?

Reply:
Once a file is loaded you don't know its content type except by knowing the path from which it was loaded.  But in this case you are simply looking for a link that is either .json or .yaml or .yml in that order.  Instead of just looking for a link that is .json I'm suggesting using the first of those that you find.  The cache routine will load the file accordingly based on its extension, using load_file in file_utils.py.  You can see other routines in there that handle different file types as needed.

Reply:
That makes sense. I was defensively coding because it is convention a file import is "the same file format" as its profile, but there is no guarantee. If I leave this check out, what happens in a JSON or YAML profiles have a XML `import/@href`, @fsuits? I meant to ask that but was not sure how you want to handle that in a clean way and with a style you all use (what is the `compliance-trestle` version of "Pythonic," treslish? haha)

Reply:
We currently don't support xml and don't plan to for upcoming releases - so if the profile requires xml processing I'm afraid it will need to error.  If this is a big limitation for certain use cases we can reassess and create an issue if there isn't one already.

Reply:
That's cool. In the context of the PR then: change the code to look for "not XML" and then not assign the array? Will this then throw the exception anyway. Or, I can be explicit about JSON/YAML.

Lemme know.

Reply:
In your current code you look for a .json link - and I'm suggesting to look for .json, then .yaml, then .yml in that order and choose the first one you find.  If you don't find any then error.  If you then hand off the url/uri for trestle to load - you can rely on it to raise an error if something isn't compatible.

This is all really just an extra thing on your current version just in case json isn't there but yaml happens to be.

Reply:
OK, updated!